x/validator-preference/msg_server.go

```diff
@@ -27,9 +29,15 @@ func (server msgServer) SetValidatorSetPreference(goCtx context.Context, msg *ty
 	preferences := msg.Preferences
 
 	// check if a user already have a validator-set created
-	existingValidator, found := server.keeper.GetValidatorSetPreference(ctx, msg.Delegator)
+	existingValidators, found := server.keeper.GetValidatorSetPreference(ctx, msg.Delegator)
+	fmt.Println(existingValidators)
 	if found {
-		preferences = existingValidator.Preferences
+		// check if the new preferences is the same as the existing preferences
+		if reflect.DeepEqual(preferences, existingValidators.Preferences) {
```

Comment:
Couldn't find a way an easier way to compare slices other than DeepEqual. Please lmk if there's any predefined compare function that i'm unaware of cc @p0mvn @mattverse @ValarDragon

Reply:
wdyt about creating a method that takes in the two val preferences as arguments, and then within the method we sort the two slices and compare the elements?